### PR TITLE
fix: compare hostname by equality

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: None
Ticket: None

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Review by mzedel — CHANGES_REQUESTED** [review]
Hm I think that's how it was before, but it won't work on staging...

**Comment by tranchitella** [comment]
> Hm I think that's how it was before, but it won't work on staging...

Maybe:

```diff --git a/src/js/components/login/login.js b/src/js/components/login/login.js
index f290fcee2..8378f68e5 100644
--- a/src/js/components/login/login.js
+++ b/src/js/components/login/login.js
@@ -84,7 +84,7 @@ export const EntryLink = ({ className = '', target = 'signup' }) => (
 );

 export const LocationWarning = () => {
-  const location = Object.entries(locations).reduce((accu, [key, value]) => (window.location.hostname == value.location ? key : accu), locations.us.key);
+  const location = Object.entries(locations).reduce((accu, [key, value]) => (value.locations.indexOf(window.location.hostname) !== -1 ? key : accu), locations.us.key);
   const { icon: Icon, title, fallback } = locationMap[location];
   return (
     <div className="flexbox centered margin-top-large">
diff --git a/src/js/constants/appConstants.js b/src/js/constants/appConstants.js
index e64d40fea..56a027fb2 100644
--- a/src/js/constants/appConstants.js
+++ b/src/js/constants/appConstants.js
@@ -35,8 +35,8 @@ export const TIMEOUTS = {
   refreshDefault: 10 * oneSecond
 };
 export const locations = {
-  eu: { key: 'eu', title: 'EU', location: 'eu.hosted.mender.io', icon: FlagEU },
-  us: { key: 'us', title: 'US', location: 'hosted.mender.io', icon: FlagUS }
+  eu: { key: 'eu', title: 'EU', locations: ['eu.hosted.mender.io'], icon: FlagEU },
+  us: { key: 'us', title: 'US', locations: ['hosted.mender.io', 'staging.hosted.mender.io'], icon: FlagUS }
 };
 export const PLANS = {
   os: {
```

File: src/js/components/login/login.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ export const EntryLink = ({ className = '', target = 'signup' }) => (
 );
 
 export const LocationWarning = () => {
-  const location = Object.entries(locations).reduce((accu, [key, value]) => (window.location.hostname.includes(value.location) ? key : accu), locations.us.key);
+  const location = Object.entries(locations).reduce((accu, [key, value]) => (window.location.hostname == value.location ? key : accu), locations.us.key);
```
Comment (mzedel) [inline src/js/components/login/login.js:87]: having a list of locations breaks other parts though... but:

```suggestion
  const location = Object.entries(locations).reduce((accu, [key, value]) => ([`staging.${value.location}`, value.location].includes(window.location.hostname) ? key : accu);
```

should get us the staging support...

Comment (tranchitella) [reply src/js/components/login/login.js]: I believe we need a start value for the reduce function to make it work.

**Review by mzedel — APPROVED** [review]

File: src/js/components/login/login.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const EntryLink = ({ className = '', target = 'signup' }) => (
 );
 
 export const LocationWarning = () => {
-  const location = Object.entries(locations).reduce((accu, [key, value]) => (window.location.hostname.includes(value.location) ? key : accu), locations.us.key);
+  const location = Object.entries(locations).reduce((accu, [key, value]) => ([`staging.${value.location}`, value.location].includes(window.location.hostname) ? key : accu), 'us');
```
Comment (mzedel) [inline src/js/components/login/login.js:87]:
```suggestion
  const location = Object.entries(locations).reduce((accu, [key, value]) => ([`staging.${value.location}`, value.location].includes(window.location.hostname) ? key : accu), locations.us.key);
```

**Pull request merged**